sympy/ntheory/factor_.py

```diff
@@ -1413,6 +1413,8 @@ def eval(cls, n):
             for p, k in factors.items():
                 t *= (p - 1) * p**(k - 1)
             return t
+        elif not n.is_Symbol:
+            raise ValueError("n must be a positive integer")
```

Comment:
@aktech 
I will update the pr in a while.
However no such test case exists in the repo. Will add the test case too.
Thanks.